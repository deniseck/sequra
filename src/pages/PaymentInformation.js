import styled from 'styled-components';
const Wrapper = styled.div`
    position: fixed;
    top: 30%;
    left: 50%;
    width: 400px;
    height: 500px;
    
    padding: 16px 32px;
    margin: 16px 0px;
    border: 1px solid;
    background-color: white;

    z-index:10;
    transform: translate(-50%,-50%);
    ${({ show }) => show && `
        display: display-block;
  `}
  ${({ show }) => !show && `
        display: none;
  `}
`;
const Header = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    margin-bottom: 32px;
    font-size: 1.5em;
`;
const Close = styled.div`
    cursor: pointer;
`;
const Content = styled.div`
p {
    margin: 32px 0px;
  }
`;
const Footer = styled.div`
margin-top: 64px;

p {
    margin: 16px 0px;
  }
`;

const PaymentInformation = (props) => {
    const { show, fee, handleClose } = props;
    return (
        <Wrapper show={show}>
            <Header>
                <div>
                    Fracciona tu pago
                </div>
                <Close onClick={handleClose}>x</Close>
            </Header>
            <Content>
                <p>1. Eliges "Fraciona tu pago" al realizar tu pedido y pagas sólo la primera cuota.</p>
                <p>2. Recibes tu pedido</p>
                <p>3. El resto de pagos se cargarán automáticamente en tu tarjeta.</p>
            </Content>
            <Footer>
                <strong>¡Así de simple!</strong>
                <div>Además, en el importe mostrado ya se incluye la cuota única mensual de {fee}/mes, por lo que no tendrás ninguna sorpresa.</div>
            </Footer>
        </Wrapper>
    )
}

export default PaymentInformation;