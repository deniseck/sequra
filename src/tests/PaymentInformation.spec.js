import PaymentInformation from '../pages/PaymentInformation';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';



it('snapshot test', () => {
    const component = renderer.create(
        <PaymentInformation show={true} fee="5 e" handleClose={() => { }}></PaymentInformation>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('Show the fee correctly', () => {
    render(
        <PaymentInformation show={true} fee="3 e" handleClose={() => { }}></PaymentInformation>
    );
    expect(screen.getByText(`Además, en el importe mostrado ya se incluye la cuota única mensual de 3 e/mes, por lo que no tendrás ninguna sorpresa.`)).toBeInTheDocument()
});