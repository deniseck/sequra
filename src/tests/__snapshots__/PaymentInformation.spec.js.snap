// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="PaymentInformation__Wrapper-sc-v57oa-0 eJnVdH"
>
  <div
    className="PaymentInformation__Header-sc-v57oa-1 jajfuj"
  >
    <div>
      Fracciona tu pago
    </div>
    <div
      className="PaymentInformation__Close-sc-v57oa-2 kPXJIx"
      onClick={[Function]}
    >
      x
    </div>
  </div>
  <div
    className="PaymentInformation__Content-sc-v57oa-3 ivrCRD"
  >
    <p>
      1. Eliges "Fraciona tu pago" al realizar tu pedido y pagas sólo la primera cuota.
    </p>
    <p>
      2. Recibes tu pedido
    </p>
    <p>
      3. El resto de pagos se cargarán automáticamente en tu tarjeta.
    </p>
  </div>
  <div
    className="PaymentInformation__Footer-sc-v57oa-4 kshnVR"
  >
    <strong>
      ¡Así de simple!
    </strong>
    <div>
      Además, en el importe mostrado ya se incluye la cuota única mensual de 
      5 e
      /mes, por lo que no tendrás ninguna sorpresa.
    </div>
  </div>
</div>
`;
